src/patternfly/components/Drawer/drawer.scss

```diff
@@ -33,58 +41,180 @@ $pf-c-drawer__panel--width-list: (25, 33, 50, 66, 75);
 
   // Divider
-  --pf-c-drawer--m-divider__panel--after--Width: var(--pf-global--BorderWidth--sm);
-  --pf-c-drawer--m-divider--m-expanded__panel--after--BackgroundColor: var(--pf-global--BorderColor--100);
-
-  display: flex;
+  --pf-c-drawer__panel--after--Width: var(--pf-global--BorderWidth--sm);
+  --pf-c-drawer__panel--after--BackgroundColor: transparent;
+  --pf-c-drawer--m-inline--m-expanded__panel--after--BackgroundColor: var(--pf-global--BorderColor--100);
+
+  // Simple list
+  --pf-c-drawer--c-simple-list__title--PaddingTop: var(--pf-global--spacer--lg);
+  --pf-c-drawer--c-simple-list__title--PaddingRight: var(--pf-global--spacer--lg);
+  --pf-c-drawer--c-simple-list__title--PaddingBottom: var(--pf-global--spacer--md);
+  --pf-c-drawer--c-simple-list__title--PaddingLeft: var(--pf-global--spacer--lg);
+
+  // Simple list item
+  --pf-c-drawer--c-simple-list__item-link--PaddingRight: var(--pf-global--spacer--lg);
+  --pf-c-drawer--c-simple-list__item-link--PaddingLeft: var(--pf-global--spacer--lg);
+
+  // Action
+  --pf-c-drawer__action--MarginTop: var(--pf-global--spacer--md);
+  --pf-c-drawer__action--MarginRight: var(--pf-global--spacer--md);
+  --pf-c-drawer__panel-body--c-drawer__action--MarginTop: calc(var(--pf-global--spacer--sm) * -1);
+  --pf-c-drawer__panel-body--c-drawer__action--MarginRight: calc(var(--pf-global--spacer--md) * -1);
+  --pf-c-drawer__content-body--m-padding--c-drawer__action--MarginTop: calc(var(--pf-global--spacer--sm) * -1);
+  --pf-c-drawer__content-body--m-padding--c-drawer__action--MarginRight: calc(var(--pf-global--spacer--md) * -1);
+
+  display: grid;
+  grid-template-areas: "header" "main";
+  grid-template-rows: max-content auto;
+  grid-template-columns: 100%;
+  min-width: var(--pf-c-drawer__panel--MinWidth);
   height: 100%;
   overflow-x: hidden;
+
+  .pf-c-simple-list__title {
+    --pf-c-simple-list__title--PaddingTop: var(--pf-c-drawer--c-simple-list__title--PaddingTop);
+    --pf-c-simple-list__title--PaddingRight: var(--pf-c-drawer--c-simple-list__title--PaddingRight);
+    --pf-c-simple-list__title--PaddingBottom: var(--pf-c-drawer--c-simple-list__title--PaddingBottom);
+    --pf-c-simple-list__title--PaddingLeft: var(--pf-c-drawer--c-simple-list__title--PaddingLeft);
+  }
+
+  .pf-c-simple-list__item-link {
+    --pf-c-simple-list__item-link--PaddingRight: var(--pf-c-drawer--c-simple-list__item-link--PaddingRight);
+    --pf-c-simple-list__item-link--PaddingLeft: var(--pf-c-drawer--c-simple-list__item-link--PaddingLeft);
+  }
+}
+
+// Header area
+.pf-c-drawer__header {
+  grid-area: header;
+}
+
+// Main area
+.pf-c-drawer__main {
+  display: flex;
+  grid-area: main;
+  overflow: hidden;
 }
 
 // Content / panel
 .pf-c-drawer__content,
 .pf-c-drawer__panel {
+  display: flex;
+  flex-direction: column;
   flex-shrink: 0;
   overflow: auto;
 }
 
+// Action
+.pf-c-drawer__content-body,
+.pf-c-drawer__panel-body {
+  .pf-c-drawer__action {
+    float: right;
```

Comment:
This could create a layout like this where text wraps around the button. like this

<img width="734" alt="Screen Shot 2020-02-06 at 3 05 09 PM" src="https://user-images.githubusercontent.com/35148959/73978692-76c30800-48f2-11ea-93a7-692bd06931bb.png">


Typically whatever is to the left of the button stays to the left of the button, like the modal or popover here

<img width="432" alt="Screen Shot 2020-02-06 at 3 09 39 PM" src="https://user-images.githubusercontent.com/35148959/73978859-ca355600-48f2-11ea-8508-026124c9f9d1.png">
<img width="469" alt="Screen Shot 2020-02-06 at 3 08 52 PM" src="https://user-images.githubusercontent.com/35148959/73978863-cb668300-48f2-11ea-8a92-acd71ceb9bfe.png">

what do you think @mcarrano @mceledonia?

Reply:
I updated this. The developer will be responsible for placing the close button as some instances control the panel outside of the panel, possibly in the `__content` or in page header.